functorch/csrc/BatchRulesViews.cpp

```diff
@@ -152,7 +153,10 @@ std::tuple<Tensor,optional<int64_t>> _unsafe_view_batch_rule(
 }
 
 Tensor trace_decomp(const Tensor& self) {
-  return at::sum(at::diagonal(self));
+  static auto * trace_exec = torch::jit::GetDecompositionExecutor("aten::trace(Tensor self) -> Tensor");
```

Comment:
cc @ezyang do you know if there's any way to easily blug in boxed fallback here ?

Reply:
nvm, @horace pointed me to https://github.com/pytorch/functorch/blob/main/functorch/csrc/BatchRulesHelper.h#L114

Reply:
Actually... pointers still helpful